extensions/proposals/OES_EGL_image_external/extension.xml

```diff
@@ -44,13 +45,20 @@
 
   <idl xml:space="preserve">
 [NoInterfaceObject]
+interface WebGLVideoFrameInfo {
+  readonly attribute double currentTime;
+  readonly attribute unsigned long textureWidth;
+  readonly attribute unsigned long textureHeight;
+};
+
+[NoInterfaceObject]
 interface OES_EGL_image_external {
     const GLenum TEXTURE_EXTERNAL_OES             = 0x8D65;
     const GLenum SAMPLER_EXTERNAL_OES             = 0x8D66;
     const GLenum TEXTURE_BINDING_EXTERNAL_OES     = 0x8D67;
     const GLenum REQUIRED_TEXTURE_IMAGE_UNITS_OES = 0x8D68;
 
-    [RaisesException] void EGLImageTargetTexture2DOES(
+    [RaisesException] WebGLVideoFrameInfo EGLImageTargetTexture2DOES(
```

Comment:
I've studied Chrome's media implementation a bit and it looks like this form of the API can work. The last video frame returned by the decoder can be held on to indefinitely. If you're OK with calling this explicitly once per frame, then this is a pretty simple solution to the problem of knowing exactly which frame from the video was latched in to the TEXTURE_EXTERNAL_OES texture.